CMakeLists.txt

```diff
@@ -66,6 +66,14 @@ elseif(MSVC_VERSION LESS 1900)
    message(FATAL_ERROR "Vc 1.x requires at least Visual Studio 2015.")
 endif()
 
+if(MSVC AND (NOT DEFINED Vc_USE_MSVC_SSA_OPTIMIZER_DESPITE_BUGGY_EXP OR NOT Vc_USE_MSVC_SSA_OPTIMIZER_DESPITE_BUGGY_EXP) AND
+   CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 19.20)
+   # bug report: https://developercommunity.visualstudio.com/t/AVX-codegen-bug-on-Vc-with-MSVC-2019/1470844#T-N1521672
+   message(STATUS "WARNING! MSVC starting with 19.20 uses a new optimizer that has a bug causing Vc::exp() to return slighly wrong results.\
+ You can set Vc_USE_MSVC_SSA_OPTIMIZER_DESPITE_BUGGY_EXP=ON to still use the new optimizer on the affected MSVC versions.")
+   AddCompilerFlag("/d2SSAOptimizer-" CXX_RESULT _ok CXX_FLAGS CMAKE_CXX_FLAGS)
+endif()
+
```

Comment:
There has not been any activity on the bug report, and I had this change lying in my working copy. I think it is a reasonable way forward to disable the offending optimizer for now, but to also provide an opt-in escape hatch. Maybe we should have this is a CMake option, but then it is a very specific flag and I don't want to present it too much in cmake-gui or ccmake. What do people think?

Reply:
Let's be conservative and disable the optimizer. Correctness is more important than performance here.